### PR TITLE
Add types to constructors and private/final/internal methods (Batch II)

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | #32179, #33228
| License       | MIT
| Doc PR        | N/A

Followup to #33519, this time with:
* Form
* HttpClient
* HttpKernel
* intl
* Ldap
* Ldap
* Lock
* Messenger
* Processor
* PropertyInfo
* Routing
* Security
* Serializer
* Stopwatch
* Translation

### CODE REVIEW

File: src/Symfony/Component/Form/Command/DebugCommand.php [hunk 1]
```diff
@@ -195,13 +195,13 @@ private function getFqcnTypeClass(InputInterface $input, SymfonyStyle $io, strin
...
 
-    private function getCoreTypes()
+    private function getCoreTypes(): array
     {
         $coreExtension = new CoreExtension();
         $loadTypesRefMethod = (new \ReflectionObject($coreExtension))->getMethod('loadTypes');
         $loadTypesRefMethod->setAccessible(true);
         $coreTypes = $loadTypesRefMethod->invoke($coreExtension);
-        $coreTypes = array_map(function (FormTypeInterface $type) { return \get_class($type); }, $coreTypes);
+        $coreTypes = array_map(function (FormTypeInterface $type): string { return \get_class($type); }, $coreTypes);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Command/DebugCommand.php]: I don't think adding these to internal closures provide anything

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php [hunk 2]
```diff
@@ -94,8 +94,8 @@ public function reverseTransform($value)
         return $dateInterval;
     }
 
-    private function isISO8601(string $string)
+    private function isISO8601(string $string): bool
     {
-        return preg_match('/^P(?=\w*(?:\d|%\w))(?:\d+Y|%[yY]Y)?(?:\d+M|%[mM]M)?(?:(?:\d+D|%[dD]D)|(?:\d+W|%[wW]W))?(?:T(?:\d+H|[hH]H)?(?:\d+M|[iI]M)?(?:\d+S|[sS]S)?)?$/', $string);
+        return (bool) preg_match('/^P(?=\w*(?:\d|%\w))(?:\d+Y|%[yY]Y)?(?:\d+M|%[mM]M)?(?:(?:\d+D|%[dD]D)|(?:\d+W|%[wW]W))?(?:T(?:\d+H|[hH]H)?(?:\d+M|[iI]M)?(?:\d+S|[sS]S)?)?$/', $string);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php]: should be removed, the return type already handle the cast

**Comment by nicolas-grekas** [comment]
Thank you. I'm not sure about any of the changes on tests: they provide nothing quality-wise, and make merges more conflict-prone.

File: src/Symfony/Component/Form/Command/DebugCommand.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/Form/Command/DebugCommand.php]: Reverted. TBH, the whole closure feels unnecessary. Why aren't we using `'get_class'` as callable here? 🤔

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php]: Fair enough, reverted.

**Comment by derrabus** [comment]
> Thank you. I'm not sure about any of the changes on tests: they provide nothing quality-wise, and make merges more conflict-prone.

On the last PR, we agreed that I could add types to tests that don't exist on the 3.4 branch. That should be the case for the Messenger and HttpClient tests that I've patched here.

File: src/Symfony/Component/Intl/Util/GitRepository.php [hunk 3]
```diff
@@ -85,12 +85,12 @@ public function checkout($branch)
         $this->execInPath(sprintf('git checkout %s', escapeshellarg($branch)));
```
Comment (Tobion) [inline src/Symfony/Component/Intl/Util/GitRepository.php]: the $branch parameter could be string typed as far as I see

**Review by nicolas-grekas — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @derrabus.

**Pull request merged**